irods/test/data_obj_test.py

```diff
@@ -1691,6 +1693,29 @@ def replica_number_from_resource_name(data_path, resc):
             if dir1: self.sess.resources.get(uniq1).remove()
             if dir2: self.sess.resources.get(uniq2).remove()
 
+    def _auto_close_test(self, name, content):
+        d = None
+        try:
+            d = self.sess.data_objects.get(name)
+            self.assertEqual(int(d.replicas[0].status), 1)
+            self.assertEqual(d.open('r').read().decode(), content)
+        finally:
+            if d: d.unlink(force = True)
```

Comment:
Should the `else` case be considered an error and/or should we be asserting that the data object exists?

Reply:
Not sure, please clarify ....
   - in this context, the data object will exist or the test will throw and indicate the data_objects.get line as the issue. In that scenario, either the program logic or iRODS itself is fundamentally flawed at compile time(?)
   - what is precisely the `else` case in question ... the `finally`, or the failure of the assertions to be `True` ?

Reply:
Oh! Yes, that makes sense. Good stuff :)